sqlglot/parser.py

```diff
@@ -3880,6 +3881,25 @@ def _parse_merge(self) -> exp.Expression:
             expressions=whens,
         )
 
+    def _parse_pragma(self) -> t.Optional[exp.Expression]:
+        this = self._parse_primary() or self._parse_id_var()
+        schema = None
+        value = None
+        func = None
+
+        while self._match(TokenType.DOT):
+            schema = this
+            this = self._parse_primary() or self._parse_id_var()
+
+        if self._match(TokenType.EQ):
```

Comment:
I think we can dry this out to

```python
func = self._match(TokenType.L_PAREN)
self._match(TokenType.EQ)
value = self._parse_var_or_string() or self._parse_number() or self._parse_id_var()
self._match(TokenType.R_Paren)
```

Reply:
Changed it the way you suggested, but personally, I think the readability suffers.

Reply:
Thanks for making the changes, happy to revert it later on if Toby agrees about the readability issue. It's a common pattern to dry out things like this in the codebase, so I'm probably biased towards this style more.